### ISSUE TITLE
[BUG] can't upgrade flutter

### ISSUE THREAD

**Issue by jamesdixon** [leoafarias/fvm#154]
**Before creating a bug report please make check the following**
- You have installed fvm using `pub global activate fvm` not `flutter pub...`
- If you have used flutter. Please install correctly, run `pub cache repair`. Close the terminal and try again.
- If you are on Windows. Make sure you are running the terminal as `administrator` or with `developer` permissions when using the `fvm use <version>` command.

**Describe the bug**
Attempted to upgrade my Flutter SDK to 1.22 from 1.20. I first installed `1.22.0` via `fvm install 1.22.0` and then used `fvm flutter upgrade`. I get the message `You are not currently on a release branch. Use git to check out an official branch ('stable', 'beta', 'dev', or 'master') and retry, for example:
  git checkout stable`

**To Reproduce**
1. Go to terminal..
2.Run `fvm install 1.22.0`
3. Run `fvm use 1.22.0`
4. Run `fvm flutter upgrade` 
4. See error `You are not currently on a release branch. Use git to check out an official branch ('stable', 'beta', 'dev', or 'master') and retry, for example:
  git checkout stable`

**Expected behavior**
Expected to be able to upgrade.

**Logs**
```
[ +121 ms] executing: [/Users/jamesdixon/fvm/versions/1.22.0/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +43 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] d408d302e22179d598f467e11da5dd968dbdc9ec
[        ] executing: [/Users/jamesdixon/fvm/versions/1.22.0/] git tag --points-at HEAD
[  +27 ms] Exit code 0 from: git tag --points-at HEAD
[        ] 1.22.0
           1.22.0-12.3.pre
[  +23 ms] executing: [/Users/jamesdixon/fvm/versions/1.22.0/] git rev-parse --abbrev-ref HEAD
[  +10 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] HEAD
[        ] executing: [/Users/jamesdixon/fvm/versions/1.22.0/] git rev-parse --abbrev-ref --symbolic @{u}
[   +9 ms] Exit code 128 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] fatal: HEAD does not point to a branch
[  +38 ms] executing: [/Users/jamesdixon/fvm/versions/1.22.0/] git tag --points-at HEAD
[  +31 ms] Exit code 0 from: git tag --points-at HEAD
[        ] 1.22.0
           1.22.0-12.3.pre
[   +5 ms] executing: [/Users/jamesdixon/fvm/versions/1.22.0/] git fetch --tags
[ +885 ms] executing: [/Users/jamesdixon/fvm/versions/1.22.0/] git rev-parse --verify @{u}
[  +13 ms] fatal: HEAD does not point to a branch
[  +14 ms] "flutter upgrade" took 955ms.
[   +4 ms] You are not currently on a release branch. Use git to check out an official branch ('stable', 'beta', 'dev', or 'master') and retry, for example:
             git checkout stable
[   +2 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
           #1      UpgradeCommandRunner.fetchRemoteRevision (package:flutter_tools/src/commands/upgrade.dart:225:9)
           <asynchronous suspension>
           #2      UpgradeCommandRunner.runCommandFirstHalf (package:flutter_tools/src/commands/upgrade.dart:103:43)
           #3      UpgradeCommandRunner.runCommand (package:flutter_tools/src/commands/upgrade.dart:85:13)
           #4      UpgradeCommand.runCommand (package:flutter_tools/src/commands/upgrade.dart:60:27)
           #5      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:972:18)
           #6      _rootRunUnary (dart:async/zone.dart:1198:47)
           #7      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #8      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #9      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #10     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #11     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #12     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
           #13     _rootRun (dart:async/zone.dart:1190:13)
           #14     _CustomZone.run (dart:async/zone.dart:1093:19)
           #15     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
           #16     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
           #17     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
           #18     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
           #19     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
           #20     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)


[   +1 ms] ensureAnalyticsSent: 0ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
```

**Additional context**
When I run `fvm flutter --version`, I see the proper version but `channel` is unknown.

```
Flutter 1.22.0 • channel unknown • unknown source
Framework • revision d408d302e2 (3 days ago) • 2020-09-29 11:49:17 -0700
Engine • revision 5babba6c4d
Tools • Dart 2.10.0
```

Thanks in advance!

**Comment by jamesdixon** [comment]
Also tried the pre-release version of `fvm` (`1.3.0-dev.4`) but no luck. Maybe my order of commands is off?

**Comment by leoafarias** [comment]
@jamesdixon the upgrade is a functionality of the channel. You are currently installing and pinning a release itself.

You should run 'fvm install beta' for example or 'fvm install master' once you are in the channel you will be able to upgrade.

Once way is pinning the 'release channel' the other is the 'release' itself.

Let me know if that makes sense.

Thanks

**Comment by jamesdixon** [comment]
Hi Leo,

Thanks! That worked as suggested. My apologies if I've wasted your time.

Thanks again for the great project 👍 

Best

**Comment by leoafarias** [comment]
No worries, thanks for the well descriptive issue.

**Comment by tony123S** [comment]
It doesn't work on my case.

```
xxx@xxx-Inspiron-14-xxx:~/Desktop/xxx$ fvm flutter --version
Flutter 2.2.1 • channel unknown • unknown source
Framework • revision 02c026b03c (4 months ago) • 2021-05-27 12:24:44 -0700
Engine • revision 0fdb562ac8
Tools • Dart 2.13.1
```

Then I install stable
```

xxx@xxx-Inspiron-14-xxx:~/Desktop/xxx$ fvm install stable
Flutter 2.5.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision ffb2ecea52 (12 days ago) • 2021-09-17 15:26:33 -0400
Engine • revision b3af521a05
Tools • Dart 2.14.2



___________________________________________________


Update Available 2.0.5 → 2.2.3 
Run pub global activate fvm to update
Changelog: https://pub.dev/packages/fvm/changelog

___________________________________________________


```

Upgrade it

```
xxx@xxx-Inspiron-14-xxx:~/Desktop/xxx$ fvm flutter upgrade
You are not currently on a release branch. Use git to check out an official branch ('stable', 'beta', 'dev', or 'master') and retry, for example:
  git checkout stable
```

Check version again

```
xxx@xxx-Inspiron-14-xxx:~/Desktop/xxx$ fvm flutter --version
Flutter 2.2.1 • channel unknown • unknown source
Framework • revision 02c026b03c (4 months ago) • 2021-05-27 12:24:44 -0700
Engine • revision 0fdb562ac8
Tools • Dart 2.13.1
```
